### DISCUSSION TITLE
Postfx ui + shader pack

### DISCUSSION THREAD

**Discussion by Jigoku** [redeclipse/base#1208] [Maps and Content Showcase]
Here's a ui for the post processing effects, with support for adding/changing shader parameters quick and easily. I've also added 28 new effects, most of which were ported from MIT licensed shaders available on github with some changes to correct for resolution scaling and other misc tweaks.

![2020-06-28-164724_1920x1080_scrot](https://user-images.githubusercontent.com/1535179/85952420-0bd9a280-b961-11ea-80b4-4bbb87e5f658.png)

Additional postfx shaders are:
* Brightness/Contrast
* Bleach Bypass
* Convergence
* CRT
* Cut & Slide
* Denoise
* Emboss
* Film
* Filmic
* Frei-Chen Edge Detection
* Heat Haze
* High Pass
* Infrared
* Tilt Shift (Horizontal)
* Tilt Shift (Vertical)
* Noise Warp
* Pixelate
* Posterize
* Reinhard
* Reinhard (Luma)
* Reinhard (White Luma)
* RGB Offset
* RomBinDaHouse
* Sepia
* Shake
* Unreal
* Vibrance
* Vignette

Instructions to try this out:

1. Download [ui_postfx.cfg](https://gist.github.com/Jigoku/81cccf022588e44af9ad5a4e2306df94) to your local data folder
2. Add `exec ui_postfx.cfg` to your ***autoexec.cfg***

3. Download [shaders.cfg](https://gist.github.com/Jigoku/aaeff68bcb6fce519d0cfa87b74faf88) to your local data folder
4. Open up ***config/glsl/postfx.cfg*** in the game folder
5. At the very bottom of the file write `exec shaders.cfg`
6. Run `/showpostfx` in game to access the ui. 

(Thanks to quin/dziq/q009/razgriz for helping with my annoying questions :zany_face: )

Some screenshots of the included shader combinations:
![20200628083022](https://user-images.githubusercontent.com/1535179/85952644-a686b100-b962-11ea-8ca7-0e76f52d1f59.png)
![20200628165856](https://user-images.githubusercontent.com/1535179/85952604-67f0f680-b962-11ea-9216-7416c2b2b695.png)
![20200628165937](https://user-images.githubusercontent.com/1535179/85952606-69222380-b962-11ea-8eee-1d496e17c0dc.png)
![20200628083332](https://user-images.githubusercontent.com/1535179/85952645-aa1a3800-b962-11ea-9835-948e04466b55.png)
![20200628165307](https://user-images.githubusercontent.com/1535179/85952599-6293ac00-b962-11ea-8622-c2ede486ed47.png)
![20200628165440](https://user-images.githubusercontent.com/1535179/85952602-64f60600-b962-11ea-98ce-a1c92562e34a.png)
![20200628172827](https://user-images.githubusercontent.com/1535179/85952969-d6cf4f00-b964-11ea-84b0-be98e4e5cf85.png)

**Comment by Jigoku** [comment]
Just updated the ***ui_postfx.cfg*** link, as it was incorrectly named shaders.cfg

**Comment by Jigoku** [comment]
Added 3 shaders:
* Emboss
* Infrared
* Vibrance

**Comment by Jigoku** [comment]
Added 6 shaders:

* Filmic
* Reinhard
* Reinhard (Luma)
* Reinhard (White Luma)
* RomBinDaHouse
* Unreal

These are all tone mapping effects.

**Comment by Jigoku** [comment]
Added 2 shaders:

* Posterize
* CRT

**Comment by Ulukai55** [comment]
Wow this looks cool. Would be interesting to see if it can be used in-game (post-fx randomizer mutator or something), or as an effect when close to an explosion e.g.

**Comment by Ulukai55** [comment]
Maybe similar to an arms-race game mode, there could be a mutator that changes your postfx filter after every kill.
